From: Hannes Reinecke <hare@suse.de>
Date: Fri, 26 Feb 2021 08:17:27 +0100
Subject: nvme-fc: set NVME_REQ_CANCELLED in nvme_fc_terminate_exchange()
Patch-mainline: v5.12-rc3
Git-commit: 3c7aafbc8d3d4d90430dfa126847a796c3e4ecfc
References: bsc#1184259

nvme_fc_terminate_exchange() is being called when exchanges are
being deleted, and as such we should be setting the NVME_REQ_CANCELLED
flag to have identical behaviour on all transports.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Reviewed-by: Keith Busch <kbusch@kernel.org>
Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
Reviewed-by: James Smart <jsmart2021@gmail.com>
Reviewed-by: Daniel Wagner <dwagner@suse.de>
Signed-off-by: Christoph Hellwig <hch@lst.de>
---
 drivers/nvme/host/fc.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/nvme/host/fc.c
+++ b/drivers/nvme/host/fc.c
@@ -2440,6 +2440,7 @@ nvme_fc_terminate_exchange(struct reques
 	struct nvme_fc_ctrl *ctrl = to_fc_ctrl(nctrl);
 	struct nvme_fc_fcp_op *op = blk_mq_rq_to_pdu(req);
 
+	op->nreq.flags |= NVME_REQ_CANCELLED;
 	__nvme_fc_abort_op(ctrl, op);
 	return true;
 }
